File: Models/EquipmentModelStateHourlyEarnings.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API_CRUD_Test.Models
{
    public class EquipmentModelStateHourlyEarnings
    {
       
        public float Value { get; set; }
        
        [ForeignKey("EquipmentState")]
        public Guid EquipmentStateId { get; set; } //chave estrangeira
        
        [ForeignKey("EquipmentModel")]
        public Guid EquipmentModelId { get; set; } //chave estrangeira

     
    }
}

File: testebackendestagiov3/src/main/java/machado/antonio/testebackendestagiov3/entities/EquipmentModelStateHourlyEarnings.java
package machado.antonio.testebackendestagiov3.entities;

import org.springframework.stereotype.Component;

import jakarta.persistence.Column;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Component
@Entity
@Table(name = "equipment_model_state_hourly_earnings", schema = "operation")
public class EquipmentModelStateHourlyEarnings {

	@EmbeddedId
	private EquipmentModelStateHourlyEarningsId id;

	@ManyToOne
	@JoinColumn(name = "equipment_model_id", insertable = false, updatable = false)
	private EquipmentModel equipmentModel;

	@ManyToOne
	@JoinColumn(name = "equipment_state_id", insertable = false, updatable = false)
	private EquipmentState equipmentState;

	@Column(nullable = false)
	private Double value;

	// Getters and Setters:

	public EquipmentModelStateHourlyEarningsId getId() {
		return id;
	}

	public void setId(EquipmentModelStateHourlyEarningsId id) {
		this.id = id;
	}

	public Double getValue() {
		return value;
	}

	public void setValue(Double value) {
		this.value = value;
	}

}
